src/scheduler.cc

```diff
@@ -4,13 +4,25 @@
 
 void Scheduler::RegisterProcess(Process& proc) {
   using Status = Process::Status;
+  lock_.Lock();
   assert(number_of_process_ < kNumberOfProcess);
   assert(proc.GetStatus() == Process::Status::kNotScheduled);
+  for (int i = 0; i < number_of_process_; i++) {
+    if (process_[i]->GetStatus() != Status::kStopped)
+      continue;
+    delete (Process*)process_[i];
```

Comment:
The object pointed by `process_[i]` was not allocated by `new` operator, so using delete on the object may cause an error. For now, liumOS doesn't support freeing memory, so please just remove this line (or leave a TODO comment here).